ext/curl/interface.c

```diff
@@ -416,6 +422,19 @@ PHP_MINIT_FUNCTION(curl)
 }
 /* }}} */
 
+/* {{{ PHP_GINIT_FUNCTION */
+PHP_GINIT_FUNCTION(curl)
+{
+	zend_hash_init(&curl_globals->persistent_share_handles, 0, NULL, curl_share_free_persistent, true);
+	GC_MAKE_PERSISTENT_LOCAL(&curl_globals->persistent_share_handles);
+}
+/* }}} */
+
+PHP_GSHUTDOWN_FUNCTION(curl)
+{
+	zend_hash_destroy(&curl_globals->persistent_share_handles);
```

Comment:
I'm wondering if it would make sense to explicitly close persistent shares with `curl_share_close`? It's a no-op for regular shares, but it may be useful for persistent ones?